### PR TITLE
Fix: markVariableAsUsed() should work in Node.js env (fixes #2089)

### LINKED ISSUE #2089: api.markVariableAsUsed does not check Node.js scope
The Node.js scope is not checked by the api.markVariableAsUsed function making these variables impossible to mark as used.

```javascript
"use strict";

var eslint = require("./lib/api.js").linter;

var code = "var a = 1, b = 2;";

eslint.reset();
eslint.on("Program:exit", function() {
    eslint.markVariableAsUsed("a"); // return false, the variable was not found in the scope
});

eslint.verify(code, { env: { node: true }}, "test.js", true);
```

### PR DESCRIPTION
**Pull request by nzakas**

### CODE REVIEW

File: lib/eslint.js [hunk 1]
```diff
@@ -998,6 +998,11 @@ module.exports = (function() {
             i,
             len;
 
+        // Special Node.js scope means we need to start one level deeper
+        if (scope.type === "global" && currentConfig.ecmaFeatures.globalReturn) {
```
Comment (yannickcr) [inline lib/eslint.js]: We also need to check this scope is the ecmaFeature `modules` is used.

I think the condition should be:

```javascript
 if (scope.type === "global" (currentConfig.ecmaFeatures.globalReturn || currentConfig.ecmaFeatures.modules)) {
```

**Comment by glenjamin** [comment]
Anything I can do to help get this in?

File: lib/eslint.js (hunk 1 above)
Comment (nzakas) [inline lib/eslint.js]: Can you explain why?

Comment (yannickcr) [inline lib/eslint.js]: Not really (I do not clearly understand how these scopes are working :( ), I just noticed we got the same behavior when we have `globalReturn` or `modules` to `true`: we need to check `scope.childScopes[0]` to get all the variables.

Test to reproduce the issue:

```js
var code = "var a = 1, b = 2;";

eslint.reset();
eslint.on("Program:exit", function() {
	var globalScope = eslint.getScope(),
	childScope = globalScope.childScopes[0];

	eslint.markVariableAsUsed("a");

	assert.isTrue(getVariable(childScope, "a").eslintUsed);
	assert.isUndefined(getVariable(childScope, "b").eslintUsed);
});

eslint.verify(code, { ecmaFeatures: { modules: true }}, filename, true);
```

Comment (nzakas) [inline lib/eslint.js]: Can you open a separate issue about that? I'm not comfortable making such a change until I understand what's going on.

Comment (yannickcr) [inline lib/eslint.js]: ok, no problem :)

**Pull request merged**